backend/src/user/user.repository.ts

```diff
@@ -10,8 +10,16 @@ export class UserRepository {
     return await this.prisma.user.findMany()
   }
 
-  async findOne(id: number) {
-    return await this.prisma.user.findUnique({ where: { id } })
+  async findOne(searchQuery) {
+    return searchQuery.id
```

Comment:
Tu mam duze watpliwosci - searchQuery jest nieotypowane a kod na 100% zaklada ze bedzie tam albo id albo email. Przy takim zalozeniu poprawny typ to `{id:number} | {email:string}` i tak naprawde mozna go wprost do where wprowadzic.

Z drugiej strony ja bym chyba wydzielil to na dwie osobne metody. findById i findByEmail